feature-libs/cart/quick-order/core/services/quick-order.service.ts

```diff
@@ -152,6 +168,92 @@ export class QuickOrderService implements QuickOrderFacade {
     );
   }
 
+  /**
+   * Add deleted entry
+   */
+  addDeletedEntry(entry: OrderEntry, clearTimeout: boolean = true): void {
+    const deletedEntries = this.deletedEntries$.getValue() || {};
+    const productCode = entry.product?.code;
+
+    if (productCode) {
+      deletedEntries[productCode] = entry;
+
+      this.deletedEntries$.next(deletedEntries);
+
+      if (clearTimeout) {
+        const subscription: Subscription = timer(
+          this.deletionClearTimeout
+        ).subscribe(() => {
+          if (productCode) {
+            this.clearDeletedEntry(productCode);
+          }
+        });
+        this.clearMessageTimeouts[productCode] = subscription;
+      }
+    }
+  }
+
+  /**
+   * Return deleted entries
+   */
+  getDeletedEntries(): Observable<DeletedEntriesObject> {
+    return this.deletedEntries$;
+  }
+
+  /**
+   * Undo deleted entry
+   */
+  undoDeletedEntry(productCode: string): void {
+    this.cleanUpDeletedEntry(productCode, true);
+  }
+
+  /**
+   * Clear deleted entry from the list
+   */
+  clearDeletedEntry(productCode: string): void {
+    this.cleanUpDeletedEntry(productCode);
+  }
+
+  /**
+   * Clear all deleted entry timeout subscriptions
+   */
+  clearTimeoutSubscriptions(): void {
```

Comment:
Shouldn't it be protected?

Reply:
no as I am using it also in ngOnDestroy in quick order component